src/scheduler.cc

```diff
@@ -327,32 +367,32 @@ void SchedulerService::schedule() {
 // assign_task assumes that computation_graph_lock_ has been acquired.
 // assign_task assumes that the canonical objrefs for its arguments are all ready, that is has_canonical_objref() is true for all of the call's arguments
 void SchedulerService::assign_task(OperationId operationid, WorkerId workerid) {
-  ObjStoreId objstoreid = get_store(workerid);
-  const Task& task = computation_graph_.get_task(operationid);
   ClientContext context;
   ExecuteTaskRequest request;
   ExecuteTaskReply reply;
-  RAY_LOG(RAY_INFO, "starting to send arguments");
-  for (size_t i = 0; i < task.arg_size(); ++i) {
-    if (!task.arg(i).has_obj()) {
-      ObjRef objref = task.arg(i).ref();
-      ObjRef canonical_objref = get_canonical_objref(objref);
-      {
-        // Notify the relevant objstore about potential aliasing when it's ready
-        std::lock_guard<std::mutex> alias_notification_queue_lock(alias_notification_queue_lock_);
-        alias_notification_queue_.push_back(std::make_pair(objstoreid, std::make_pair(objref, canonical_objref)));
+  std::lock_guard<std::mutex> workers_lock(workers_lock_);
+  const Task& task = computation_graph_.get_task(operationid);
+  request.mutable_task()->CopyFrom(task); // TODO(rkn): Is ownership handled properly here?
+  Status status = workers_[workerid].worker_stub->ExecuteTask(&context, request, &reply);
```

Comment:
I see that you move this further up because you need the reply in order to check `reply.rejected`. However, this means you're holding on to `workers_lock_` for a really long time, which may be introducing deadlock (there is deadlock somewhere, `runtest.py` hangs). For now, I'd suggest ruling out the possibility of a rejected reply and replace this line in `worker.cc` which rejects tasks with a fatal error.